File: src/main/java/net/finance/entity/Savings.java
package net.finance.entity;

import lombok.Data;

import javax.persistence.*;
import java.io.Serializable;
import java.math.BigDecimal;
import java.util.Date;

@Data
@Entity
@Table(name = "savings")
public class Savings implements Serializable {


    private static final long serialVersionUID = -6332118260679053853L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", unique = true, nullable = false, precision = 3, scale = 0)
    private Integer id;
    @Column(name = "description", nullable = false, length = 40)
    private String description;
    @Column(name = "value", nullable = false)
    private BigDecimal value;
    @ManyToOne(optional = false)
    @JoinColumn(name = "app_user", referencedColumnName = "id")
    private User user;
    @Column(name = "created_date", nullable = false)
    private Date createdDate;
}

File: src/main/kotlin/com/finances/entity/Savings.kt
package com.finances.entity

import java.math.BigDecimal
import java.time.LocalDateTime
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.JoinColumn
import javax.persistence.ManyToOne
import javax.persistence.Table

@Entity
@Table(name = "savings")
class Savings(
    @JoinColumn(name = "app_user", referencedColumnName = "id")
    @ManyToOne
    val user: User,
    val description: String,
    val value: BigDecimal,
    val createdDate: LocalDateTime
) {

    constructor(id: Int, user: User, description: String, value: BigDecimal, createdDate: LocalDateTime) : this(
        user,
        description,
        value,
        createdDate
    ) {
        this.id = id
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Int = 0
}
